time stamp
bool qr
string qr_name